concent_api/utils/api_view.py

```diff
@@ -19,17 +20,23 @@ def api_view(view):
     @wraps(view)
     @csrf_exempt
     def wrapper(request, *args, **kwargs):
-        if 'HTTP_CONCENT_CLIENT_PUBLIC_KEY' not in request.META:
+        if 'HTTP_CONCENT_CLIENT_PUBLIC_KEY' not in request.META or request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'] == '':
             return JsonResponse({'error': 'Concent-Client-Public-Key HTTP header is missing on the request.'}, status = 400)
 
         try:
             client_public_key = b64decode(request.META['HTTP_CONCENT_CLIENT_PUBLIC_KEY'].encode('ascii'))
-        except TypeError:
+        except (AttributeError, BinAsciiError):
```

Comment:
In what situation can an `AttributeError` be thrown here? And why `BinAsciiError` instead of `binascii.Error`? The [docs for `b64decode()`](https://docs.python.org/3.6/library/base64.html#base64.b64decode) mention only `binascii.Error`.

Also, please use `verify=True`.

Reply:
@cameel When I was implementing it `AttributeError` was already there, so this diff must be somehow malformed along the way, I will change it, and also remove comment and add `verify=True`

About BinAsciiError, please check the import part, I am importing `binascii.Error` as `BinAsciiError` because `Error` is to common name.

Reply:
Oh, OK. Maybe it would be better to do `import binascii` and then catch `binascii.Error`? I think it would be a bit less confusing. Also, it's easier to grep for stuff when it's not renamed like that.

Reply:
Right, I will update now.